test/integration/kubernetes_deploy_test.rb

```diff
@@ -1098,6 +1098,46 @@ def test_network_policies_are_deployed_first
     ], in_order: true)
   end
 
+  # We want to be sure that failures to apply resources with potentially sensitive output don't leak any content.
+  # Currently our only sensitive resource is `Secret`, but we cannot reproduce a failure scenario where the kubectl
+  # output contains the filename (which would trigger some extra logging). This test stubs `Deployment` to be sensitive
+  # to recreate such a condition
+  def test_apply_failure_with_sensitive_resources_hides_template_content
+    logger.level = 0
+    KubernetesDeploy::Deployment.any_instance.expects(:kubectl_output_is_sensitive?).returns(true).at_least_once
```

Comment:
This definitely cannot be run in parallel. `.any_instance.expects` is from `mocha`, which is not thread-safe

Reply:
Resolved.